Binary files ../../subsonic/tags/android-release-2.8/res/drawable/ic_menu_screen_option.png and ./res/drawable/ic_menu_screen_option.png differ
diff -rupN ../../subsonic/tags/android-release-2.8/res/menu/nowplaying.xml ./res/menu/nowplaying.xml
--- ../../subsonic/tags/android-release-2.8/res/menu/nowplaying.xml	2011-06-15 14:47:25.000000000 -0700
+++ ./res/menu/nowplaying.xml	2011-06-16 12:10:21.000000000 -0700
@@ -18,5 +18,10 @@
         a:icon="@drawable/ic_menu_remove_all"
         a:title="@string/download.menu_remove_all"
         />
-        
+
+    <item 
+        a:id="@+id/menu_screen_option"
+        a:icon="@drawable/ic_menu_screen_option"
+        a:title="@string/download.menu_screen_option"
+        />        
 </menu>
diff -rupN ../../subsonic/tags/android-release-2.8/res/values/strings.xml ./res/values/strings.xml
--- ../../subsonic/tags/android-release-2.8/res/values/strings.xml	2011-06-15 14:47:23.000000000 -0700
+++ ./res/values/strings.xml	2011-06-16 09:25:40.000000000 -0700
@@ -82,6 +82,7 @@
     <string name="download.menu_lyrics">Lyrics</string>
     <string name="download.menu_remove">Remove song</string>
     <string name="download.menu_remove_all">Remove all</string>
+    <string name="download.menu_screen_option">Screen On</string>
     <string name="download.menu_shuffle">Shuffle</string>
     <string name="download.menu_save">Save playlist</string>
     <string name="download.menu_shuffle_notification">Playlist was shuffled</string>
diff -rupN ../../subsonic/tags/android-release-2.8/src/net/sourceforge/subsonic/androidapp/activity/DownloadActivity.java ./src/net/sourceforge/subsonic/androidapp/activity/DownloadActivity.java
--- ../../subsonic/tags/android-release-2.8/src/net/sourceforge/subsonic/androidapp/activity/DownloadActivity.java	2011-06-15 14:47:22.000000000 -0700
+++ ./src/net/sourceforge/subsonic/androidapp/activity/DownloadActivity.java	2011-06-16 11:33:43.000000000 -0700
@@ -38,6 +38,7 @@ import android.view.MenuItem;
 import android.view.View;
 import android.view.ViewGroup;
 import android.view.Window;
+import android.view.WindowManager;
 import android.view.animation.AnimationUtils;
 import android.widget.AdapterView;
 import android.widget.ArrayAdapter;
@@ -100,6 +101,7 @@ public class DownloadActivity extends Su
     private DownloadFile currentPlaying;
     private long currentRevision;
     private EditText playlistNameView;
 
     /**
      * Called when the activity is first created.
@@ -266,6 +319,11 @@ public class DownloadActivity extends Su
         onCurrentChanged();
         onProgressChanged();
         scrollToCurrent();
+       if (getDownloadService().getScreenOff()) {
+               this.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+       } else {
+               this.getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+       }
     }

     // Scroll to current playing/downloading.
@@ -351,6 +353,12 @@ public class DownloadActivity extends Su
         boolean savePlaylistEnabled = !Util.isOffline(this);
         savePlaylist.setEnabled(savePlaylistEnabled);
         savePlaylist.setVisible(savePlaylistEnabled);
+        MenuItem screenOption = menu.findItem(R.id.menu_screen_option);
+        if (getDownloadService().getScreenOff()) {
+        	screenOption.setTitle("Screen On");
+        } else {
+        	screenOption.setTitle("Screen Off");
+        }
         return super.onPrepareOptionsMenu(menu);
     }
 
@@ -409,6 +417,15 @@ public class DownloadActivity extends Su
                 getDownloadService().clear();
                 onDownloadListChanged();
                 return true;
+            case R.id.menu_screen_option:
+               if (getDownloadService().getScreenOff()) {
+            		this.getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+                       getDownloadService().setScreenOff(false);
+            	} else {
+            		this.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+                       getDownloadService().setScreenOff(true);
+            	}
+                return true;
             case R.id.menu_shuffle:
                 getDownloadService().shuffle();
                 Util.toast(this, R.string.download_menu_shuffle_notification);
diff -rupN ../../../android-release-2.8/src/net/sourceforge/subsonic/androidapp/service/DownloadService.java ./src/net/sourceforge/subsonic/androidapp/service/DownloadService.java
--- ../../../android-release-2.8/src/net/sourceforge/subsonic/androidapp/service/DownloadService.java   2011-06-17 09:29:09.000000000 -0700
+++ ./src/net/sourceforge/subsonic/androidapp/service/DownloadService.java      2011-06-21 10:47:54.000000000 -0700
@@ -41,6 +41,10 @@ public interface DownloadService {^M
     RepeatMode getRepeatMode();^M
 ^M
     void setRepeatMode(RepeatMode repeatMode);^M
+    ^M
+    boolean getScreenOff();^M
+    ^M
+    void setScreenOff(boolean screenoff);^M
 ^M
     void clear();^M
 ^M
diff -rupN ../../../android-release-2.8/src/net/sourceforge/subsonic/androidapp/service/DownloadServiceImpl.java ./src/net/sourceforge/subsonic/androidapp/service/DownloadServiceImpl.java
--- ../../../android-release-2.8/src/net/sourceforge/subsonic/androidapp/service/DownloadServiceImpl.java       2011-06-17 10:25:53.000000000 -0700
+++ ./src/net/sourceforge/subsonic/androidapp/service/DownloadServiceImpl.java  2011-06-21 10:47:54.000000000 -0700
@@ -80,6 +80,8 @@ public class DownloadServiceImpl extends
     private static DownloadService instance;^M
     private String suggestedPlaylistName;^M
     private PowerManager.WakeLock wakeLock;^M
+    private boolean screenoff = true;^M
+^M
 ^M
     @Override^M
     public void onCreate() {^M
@@ -211,6 +213,15 @@ public class DownloadServiceImpl extends
     public void setRepeatMode(RepeatMode repeatMode) {^M
         Util.setRepeatMode(this, repeatMode);^M
     }^M
+    @Override^M
+    public boolean getScreenOff() {^M
+       return screenoff;^M
+    }^M
+    ^M
+    @Override^M
+    public void setScreenOff(boolean screenoff) {^M
+       this.screenoff = screenoff;^M
+    }^M
 ^M
     @Override^M
     public synchronized DownloadFile forSong(MusicDirectory.Entry song) {^M

